integration/src/integration/java/pl/allegro/tech/hermes/integration/FrontendAuthenticationConfigurationTest.java

```diff
@@ -0,0 +1,185 @@
+    public void setup() throws Exception {
+        ConfigFactory configFactory = new MutableConfigFactory()
+                .overrideProperty(Configs.FRONTEND_PORT, FRONTEND_PORT)
+                .overrideProperty(Configs.FRONTEND_SSL_ENABLED, false)
+                .overrideProperty(Configs.FRONTEND_AUTHENTICATION_ENABLED, true);
+
+        AuthenticationConfiguration authConfig = new AuthenticationConfiguration(
+                exchange -> true,
+                Lists.newArrayList(new BasicAuthenticationMechanism("basicAuthRealm")),
+                new SingleUserAwareIdentityManager(USERNAME, PASSWORD));
+
+        hermesFrontend = HermesFrontend.frontend()
+                .withBinding(configFactory, ConfigFactory.class)
+                .withBinding(authConfig, AuthenticationConfiguration.class)
+                .build();
+
+        hermesFrontend.start();
+
+        hermesServer = hermesFrontend.getService(HermesServer.class);
+        publisher = new HermesPublisher(FRONTEND_URL);
+
+        operations.buildTopic("someGroup", "topicWithAuthorization");
+    }
+
+    @AfterClass
+    public void tearDown() throws InterruptedException {
+        hermesFrontend.stop();
+    }
+
+    @Test
+    public void shouldAuthenticateUsingBasicAuth() throws Throwable {
+        //given
+        Map<String, String> headers = getHeadersWithAuthentication(USERNAME, PASSWORD);
+
+        //when
+        Response response = publisher.publish("someGroup.topicWithAuthorization",
+                TestMessage.of("hello", "world").body(), headers);
+
+        //then
+        assertThat(response.getStatusInfo().getFamily()).isEqualTo(SUCCESSFUL);
+    }
+
+    @Test
+    public void shouldNotAuthenticateUserWithInvalidCredentials() throws Throwable {
+        //given
+        Map<String, String> headers = getHeadersWithAuthentication(USERNAME, "someInvalidPassword");
+
+        //when
+        Response response = publisher.publish("someGroup.topicWithAuthorization",
+                TestMessage.of("hello", "world").body(), headers);
+
+        //then
+        assertThat(response.getStatus()).isEqualTo(StatusCodes.UNAUTHORIZED);
+    }
+
+    @Test
+    public void shouldNotAuthenticateUserWithoutCredentials() throws Throwable {
+        //when
+        Response response = publisher.publish("someGroup.topicWithAuthorization",
+                TestMessage.of("hello", "world").body());
+
+        //then
+        assertThat(response.getStatus()).isEqualTo(StatusCodes.UNAUTHORIZED);
+    }
+
+    private Map<String, String> getHeadersWithAuthentication(String username, String password) {
+        String credentials = username + ":" + password;
+        String token = "Basic " + Base64.encodeBase64String(credentials.getBytes(StandardCharsets.UTF_8));
+
+        Map<String, String> headers = new HashMap<>();
+        headers.put("Authorization", token);
+        return headers;
+    }
+
+    private class SingleUserAwareIdentityManager implements IdentityManager {
```

Comment:
\[Checkstyle\] ERROR: Class SingleUserAwareIdentityManager should be declared as final\.